node/hyperbahn/index.js

```diff
@@ -91,10 +92,19 @@ function HyperbahnClient(options) {
         'Must pass in top level tchannel');
     assert(options.tchannel.tracer, 'Top channel must have trace enabled');
     assert(options.serviceName, 'must pass in serviceName');
-    assert(Array.isArray(options.hostPortList),
-        'Must pass in hostPortList');
+    if (Array.isArray(options.hostPortList)) {
+        self.hostPortList = options.hostPortList;
+    } else if (typeof options.hostPortFile === 'string') {
+        var hostContent = safeSyncRead(options.hostPortFile);
+        assert(!hostContent.error, 'Failed to read the hostPortFile');
```

Comment:
This approach is ok for now; however in the future we need to an async read.

We actually want to keep polling the file within the advertise timeout period.